pilot/cmd/pilot-agent/status/server.go

```diff
@@ -64,6 +85,17 @@ func (s *Server) Run(ctx context.Context) {
 	// Add the handler for ready probes.
 	http.HandleFunc(readyPath, s.handleReadyProbe)
 
+	// TODO: we require non empty url to take over the health check. Make sure this is consistent in injector.
+	if s.appReadyURL != "" {
+		log.Infof("Pilot agent takes over readiness probe, path %v", s.appReadyURL)
```

Comment:
Does this really mean envoy takes over readiness probe? Not sure how pilot agent can do it.

Reply:
We'll have `kube-inject` code to modify podspec such the app readiness check will be re-written to pilot-agent's url.